caffe2-haskell/cpu/Foreign/Caffe2.hs

```diff
@@ -72,25 +73,25 @@ readBlob
   -> IO (Vector.Vector e)
 readBlob size elt (BlobPtr blob) =
   case elt of
-    Caffe2Float  -> fmap (Vector.map realToFrac) $ READ_TENSOR(blob,float,outBytes)
-    Caffe2Double -> fmap (Vector.map realToFrac) $ READ_TENSOR(blob,double,outBytes)
-    Caffe2Word8  -> fmap (Vector.map fromIntegral) $ READ_TENSOR(blob,uint8_t,outBytes)
-    Caffe2Word16 -> fmap (Vector.map fromIntegral) $ READ_TENSOR(blob,uint16_t,outBytes)
+    Caffe2Float  -> fmap coerce $ READ_TENSOR(blob,float,outBytes)
```

Comment:
```suggestion
    Caffe2Float  -> coerce $ READ_TENSOR(blob,float,outBytes)
```

Would this change work?

Reply:
I don't think `IO` is representable but I could try.